kernel/smp.c

```diff
@@ -120,17 +120,17 @@ static inline void smp_init_top(void *arg)
 	wait_for_start_signal(&cpu_start_flag);
 
 #ifdef CONFIG_SYS_CLOCK_EXISTS
-	if ((csc == NULL) || csc->reinit_timer) {
+	if ((arg == NULL) || csc.reinit_timer) {
```

Comment:
I don't think that we need the `arg == NULL` check as `csc` is now a local variable.

Reply:
@peter-mitsis actually, almost. In this case it's actually needed: if we just make this `if (csc.reinit_timer)` then in case `arg == NULL` it won't trigger, because `csc.reinit_timer` will be 0 too, but in fact this `if` has to be taken. The third `if` below also needs both. But you're right about the second one - there `arg != NULL` can be dropped. I'll update it, thanks